src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```diff
@@ -0,0 +1,139 @@
+        /// <param name="setter">The callback will be invoked to set a propagation key/value pair to the carrier.</param>
+        public abstract void Inject(Activity activity, object carrier, PropagatorSetterCallback setter);
+
+        /// <summary>
+        /// Extracts the value from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation trace Id and trace state from carrier.</param>
+        /// <param name="traceId">The extracted trace Id from the carrier.</param>
+        /// <param name="traceState">The extracted trace state from the carrier.</param>
+        public abstract void ExtractTraceIdAndState(object carrier, PropagatorGetterCallback getter, out string? traceId, out string? traceState);
+
+        /// <summary>
+        /// Extracts the baggage key-value pairs list from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation baggage list from carrier.</param>
+        /// <returns>Returns the extracted key-value pair list from teh carrier.</returns>
+        public abstract IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter);
+
+        /// <summary>
+        /// Get or set the process wide propagator object which used as the current selected propagator.
+        /// </summary>
+        public static TextMapPropagator Current
+        {
+            get
+            {
+                Debug.Assert(s_current is not null);
+                return s_current;
+            }
+
+            set
+            {
+                if (value is not null)
+                {
+                    s_current = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// returns the default propagator object which Current property will be initialized with.
+        /// </summary>
+        public static TextMapPropagator CreateDefaultPropagator() => LegacyPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which can propagate the trace context data using the root Activity and ignore any created child activities.
+        /// </summary>
+        public static TextMapPropagator CreatePassThroughPropagator() => PassThroughPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which suppress injecting any data to teh carriers.
+        /// </summary>
+        public static TextMapPropagator CreateNoOutputPropagator() => NoOutputPropagator.Instance;
```

Comment:
Is "Output" the right naming to use?  Would it be better to keep with the naming used elsewhere in the API, e.g. "NoInjection" or something like that?

Reply:
This was discussed before. actually, it was originally proposed with the `Inject` term instead of `Output`. 

The reason was the terms Inject and Extract are coming from the specs and the APIs has names including these terms will be used in the low-level libraries (e.g. Http Client and asp.net). It is not expected many people call such APIs. While the APIs like `CreateNoOutputPropagator` will be called for configuration which be done by users who really don't care how the propagators work and even not exposed to Inject/Extract operations. That is why it was preferred to use Output for simplicity for such users.